src/elm/Ui/Button.elm

```diff
@@ -34,8 +49,59 @@ type ThemeColor
     | Secondary_4
 
 
-button : ButtonMode -> ThemeColor -> String -> Bool -> Maybe (Html msg) -> msg -> Html msg
-button mode color text disabled icon action =
+type alias ButtonOptions msg =
```

Comment:
Bra jobba med å få disse komponentene over til builder pattern!

Litt usikker på `Options`-suffiksen som er brukt. Jeg er vant til å lage komponenter med typer som heter det samme som modulen, så `Radio`-modulen har en `Radio`-type, `Button`-modulen har en `Button`-type, osv.). Disse kan exposes der de brukes, og da er det lett å se hvilken modul de tilhører. Er det andre typer i den modulen som ikke er stertkt knyttet til navnet på modulen, så foretrekker jeg å ikke expose de - f.eks.  bruk `Button.ThemeColor`).

I tillegg foretrekker jeg at funksjonen som lager `Html` fra denne typen alltid heter `view`. Da er det ingen tvil om hva som skjer, og det følger Elms livssyklus bedre. For eksempel:

```elm
Button.init "foo"
    |> Button.setWhatever ...
    |> Button.view
```

Har man flere funksjoner som lager `Html` så kan de fortsatt være prefikset med `view`. For eksempel, i `Radio` så kunne du hatt `Radio.view` og `Radio.viewGroup`. I `Button` så tror jeg kanskje det hadde gitt mer mening å dytte `ButtonMode` og `ThemeColor` inn i `Button` (née `ButtonOptions`).

Reply:
Bra! Jeg skrev om slik at det ikke er `Options`, men ikke skrevet om hele Button API-et og Message API-et enda. Tenker det blir en egen PR etterpå: https://github.com/AtB-AS/webshop/issues/68